v8/payloads/auditLog.ts

```diff
@@ -0,0 +1,467 @@
+	GuildMFALevel,
+	GuildVerificationLevel,
+	IntegrationExpireBehavior,
+} from './guild';
+import type { APIRole } from './permissions';
+import type { APIUser } from './user';
+import type { APIWebhook } from './webhook';
+
+/**
+ * https://discord.com/developers/docs/resources/audit-log#audit-log-object-audit-log-structure
+ */
+export interface APIAuditLog {
+	webhooks: APIWebhook[];
+	users: APIUser[];
+	audit_log_entries: APIAuditLogEntry[];
+	integrations: APIGuildIntegration[];
+}
+
+/**
+ * https://discord.com/developers/docs/resources/audit-log#audit-log-entry-object-audit-log-entry-structure
+ */
+export interface APIAuditLogEntry {
+	target_id: string | null;
+	changes?: APIAuditLogChange[];
+	user_id: string | null;
+	id: string;
+	action_type: AuditLogEvent;
+	options?: APIAuditLogOptions;
+	reason?: string;
+}
+
+/**
+ * https://discord.com/developers/docs/resources/audit-log#audit-log-entry-object-audit-log-events
+ */
+export enum AuditLogEvent {
+	GUILD_UPDATE = 1,
+
+	CHANNEL_CREATE = 10,
+	CHANNEL_UPDATE,
+	CHANNEL_DELETE,
+	CHANNEL_OVERWRITE_CREATE,
+	CHANNEL_OVERWRITE_UPDATE,
+	CHANNEL_OVERWRITE_DELETE,
+
+	MEMBER_KICK = 20,
+	MEMBER_PRUNE,
+	MEMBER_BAN_ADD,
+	MEMBER_BAN_REMOVE,
+	MEMBER_UPDATE,
+	MEMBER_ROLE_UPDATE,
+	MEMBER_MOVE,
+	MEMBER_DISCONNECT,
+	BOT_ADD,
+
+	ROLE_CREATE = 30,
+	ROLE_UPDATE,
+	ROLE_DELETE,
+
+	INVITE_CREATE = 40,
+	INVITE_UPDATE,
+	INVITE_DELETE,
+
+	WEBHOOK_CREATE = 50,
+	WEBHOOK_UPDATE,
+	WEBHOOK_DELETE,
+
+	EMOJI_CREATE = 60,
+	EMOJI_UPDATE,
+	EMOJI_DELETE,
+
+	MESSAGE_DELETE = 72,
+	MESSAGE_BULK_DELETE,
+	MESSAGE_PIN,
+	MESSAGE_UNPIN,
+
+	INTEGRATION_CREATE = 80,
+	INTEGRATION_UPDATE,
+	INTEGRATION_DELETE,
+}
+
+/**
+ * https://discord.com/developers/docs/resources/audit-log#audit-log-entry-object-optional-audit-entry-info
+ */
+export interface APIAuditLogOptions {
+	/**
+	 * Present from:
+	 * - MEMBER_PRUNE
+	 */
+	delete_member_days?: string;
+	/**
+	 * Present from:
+	 * - MEMBER_PRUNE
+	 */
+	members_removed?: string;
+
+	/**
+	 * Present from:
+	 * - MEMBER_MOVE
+	 * - MESSAGE_PIN
+	 * - MESSAGE_UNPIN
+	 * - MESSAGE_DELETE
+	 */
+	channel_id?: string;
+
+	/**
+	 * Present from:
+	 * - MESSAGE_PIN
+	 * - MESSAGE_UNPIN
+	 */
+	message_id?: string;
+
+	/**
+	 * Present from:
+	 * - MESSAGE_DELETE
+	 * - MESSAGE_BULK_DELETE
+	 * - MEMBER_DISCONNECT
+	 * - MEMBER_MOVE
+	 */
+	count?: string;
+
+	/**
+	 * Present from:
+	 * - CHANNEL_OVERWRITE_CREATE
+	 * - CHANNEL_OVERWRITE_UPDATE
+	 * - CHANNEL_OVERWRITE_DELETE
+	 */
+	id?: string;
+
+	/**
+	 * Present from:
+	 * - CHANNEL_OVERWRITE_CREATE
+	 * - CHANNEL_OVERWRITE_UPDATE
+	 * - CHANNEL_OVERWRITE_DELETE
+	 */
+	type: AuditLogOptionsType;
+
+	/**
+	 * Present from:
+	 * - CHANNEL_OVERWRITE_CREATE
+	 * - CHANNEL_OVERWRITE_UPDATE
+	 * - CHANNEL_OVERWRITE_DELETE
+	 *
+	 * **Present only if the {@link APIAuditLogOptions#type entry type} is "role"**
+	 */
+	role_name?: string;
+}
+
+export enum AuditLogOptionsType {
+	Member = 'member',
```

Comment:
Are these the same in v8?

Reply:
Oh, I didn't there's another type specifically for audit logs

... it looks like v8 sends those as a number in a string (`"1"`), I suppose that's a bug

Reply:
Opened an issue: https://github.com/discord/discord-api-docs/issues/2088

I guess we should leave this until it gets a reply

not sure why that needs to be a different type though? we could just set it to OverwriteType (if it should be just a number)

Reply:
If they're the same type.. and only for overwrites, you can delete it from both v6 and v8 and use the OverwriteType enum instead
First tho we have to wait for Discord's reply on the issue